hack/tools/e2e-log-sync/main.go

```diff
@@ -0,0 +1,327 @@
+//go:build tools
+// +build tools
+
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/pkg/errors"
+	"github.com/valyala/fastjson"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+	"k8s.io/klog/v2"
+)
+
+var (
+	logBucket     = flag.String("log-bucket", "kubernetes-jenkins", "GCS Bucket to download the logs from")
+	logFolder     = flag.String("log-folder", "", "GCS folder to get the logs from")
+	logProwJobURL = flag.String("log-prow-job-url", "", "URL of the ProwJob to get the logs from")
+	lokiURL       = flag.String("loki-url", "http://localhost:3100/loki/api/v1/push", "Loki URL to push the logs to")
+)
+
+func main() {
+	flag.Parse()
+	if err := validateAndCompleteFlags(); err != nil {
+		fmt.Printf("Failed to validate flags: %v\n", err)
+		os.Exit(1)
+	}
+
+	if err := importLogs(); err != nil {
+		fmt.Printf("Failed to import logs: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func validateAndCompleteFlags() error {
+	if *logBucket == "" {
+		fmt.Printf("--log-bucket must not be empty\n")
```

Comment:
Is this true given that we are deriving it from the url if provided

Reply:
I had that validation given that it had a default value so folks hat to set a URL and unset the bucket to hit that case. But I dropped the entire flag now